modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/utils/setDataRecord.es.js

```diff
@@ -49,7 +49,7 @@ export default function setDataRecord(
 		if (
 			!edited &&
 			Liferay.ThemeDisplay.getDefaultLanguageId() === languageId &&
-			(value === '' || !value.length || value === false)
+			(value === '' || !value?.length || value === false)
```

Comment:
@alinedoleron since this PR is urgent you can send this after, but what do you think to extract this condition here to a function that does this verification to make the code more clear? For example:

```
const isValidValue (value) => {
  return  (value === '' || !value?.length || value === false)
}

...

if (!edited && Liferay.ThemeDisplay.getDefaultLanguageId() === languageId && 
isValidValue(value)) {
...
}			
			
```
This way you can change the conditions of the verification only in one place and makes the code more readable as well; but feel free to make this change in a follow up PR or not!